content/docs/handling-events.md

```diff
@@ -152,3 +152,21 @@ Inside a loop it is common to want to pass an extra parameter to an event handle
 The above two lines are equivalent, and use [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) and [`Function.prototype.bind`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind) respectively.
 
 In both cases, the `e` argument representing the React event will be passed as a second argument after the ID. With an arrow function, we have to pass it explicitly, but with `bind` any further arguments are automatically forwarded.
+
+### Event handlers with Currying
+
+We can also accomplish event handling by using function currying with arrow functions.  To do this we will need to slightly update our event handler. When the handler is invoked by the render function, the outer function is called which passes our arguments in.  The handler also returns an inner function.  The inner function is what is called when the given event is executed.  The inner function has access to the arguments in the outer function.
+
+```js
+class LoggingButton extends React.Component {
+  handleClick = (i) => (e) => {
```

Comment:
This is using an experimental syntax. We intentionally don't do this except for a few sections where we describe this syntax. It is confusing to people who are using plain ES6, and also is not guaranteed to end up in the spec.

Reply:
Fair enough.  Would you guys like me to just close this then?

Reply:
@gaearon,

I get your point, but as it stands now, the documentation is confusing and incomplete.

It says arrow syntax can cause unnecessary re-renders of child components, then immediately afterward, it shows an example using that flawed syntax to pass an argument to an event handler.

This literally begs the question, how do you avoid unnecessary re-renders *and* pass an argument?

Do you have to be an advanced user to want to know that? I don't think so.

If currying is the answer, it should be in this documentation. You don't have to explain currying, just provide an example and potentially link to different page that explains the concept.

So, I strongly urge you not to close this. This should definitely be fixed.

Reply:
There are quite a lot of examples in the FAQ: https://reactjs.org/docs/faq-functions.html


We should probably just link there.

Reply:
A link to an example would be a fine solution to me.